src/gmp/models/report/auditreport.js

```diff
@@ -0,0 +1,121 @@
+/* Copyright (C) 2024 Greenbone AG
+ *
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation, either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+import {isDefined} from '../../utils/identity';
+import {isEmpty} from '../../utils/string';
+
+import {parseDate} from '../../parser';
+
+import {parseFilter} from '../../collection/parser';
+
+import Model from '../../model';
+
+import ReportTask from './task';
+
+import {
+  parse_errors,
+  parseHosts,
+  parseOperatingSystems,
+  parseResults,
+  parseTlsCertificates,
+} from './parser';
+
+class AuditReportReport extends Model {
+  static entityType = 'auditreport';
+
+  parseProperties(element) {
+    return AuditReportReport.parseElement(element);
+  }
+
+  static parseElement(element) {
+    const copy = super.parseElement(element);
+
+    const {delta, compliance, scan_start, scan_end, task, scan, timestamp} =
+      element;
+
+    const filter = parseFilter(element);
+
+    copy.filter = filter;
+
+    copy.report_type = element._type;
+
+    delete copy.filters;
+
+    if (isDefined(compliance)) {
+      copy.compliance = {
+        filtered: compliance.filtered,
+        full: compliance.full,
+      };
+    }
+
+    copy.task = ReportTask.fromElement(task);
+
+    copy.results = parseResults(element, filter);
+
+    copy.hosts = parseHosts(element, filter);
+
+    copy.tlsCertificates = parseTlsCertificates(element, filter);
+
+    delete copy.host;
+
+    copy.operatingsystems = parseOperatingSystems(element, filter);
+
+    copy.errors = parse_errors(element, filter);
+
+    copy.scan_start = parseDate(scan_start);
```

Comment:
I prefer to keep `scan_start`, `scan_end` and `scan_run_status` unchanged for now since these are the same names used in reports and both audit reports and reports share some components that make use of them (for example: Summary and ToolBarIcons).